crates/trie/sparse/src/state.rs

```diff
@@ -20,6 +20,40 @@ use reth_trie_common::{
 };
 use tracing::trace;
 
+/// Provides type-safe re-use of cleared [`SparseStateTrie`]s, which helps to save allocations
+/// across payload runs.
+#[derive(Debug)]
+pub struct ClearedSparseStateTrie<
+    A = SerialSparseTrie, // Account trie implementation
+    S = SerialSparseTrie, // Storage trie implementation
+>(SparseStateTrie<A, S>);
+
+impl<A, S> From<SparseStateTrie<A, S>> for ClearedSparseStateTrie<A, S>
+where
+    A: SparseTrieInterface,
+    S: SparseTrieInterface,
+{
+    fn from(mut trie: SparseStateTrie<A, S>) -> Self {
```

Comment:
I'd prefer to have this as `fn cleared(SparseStateTrie<A, S>)` instead, which should make it clear how the sparsestatetrie is converted